core/src/main/java/io/questdb/TelemetryJob.java

```diff
@@ -0,0 +1,174 @@
+                compiler.compile("CREATE TABLE " + tableName + " (ts timestamp, event short)", sqlExecutionContext);
+            }
+
+            if (getTableStatus(path, configTableName) == TableUtils.TABLE_DOES_NOT_EXIST) {
+                compiler.compile("CREATE TABLE " + configTableName + " (id long256, enabled boolean)", sqlExecutionContext);
+            }
+        }
+
+        try {
+            this.writer = new TableWriter(cairoConfig, tableName);
+        } catch (CairoException ex) {
+            LOG.error().$("could not open [table=").utf8(tableName).$("]").$();
+            throw ex;
+        }
+
+        // todo: close writerConfig. We currently keep it opened to prevent users from modifying the table.
+        // Once we have a permission system, we can use that instead.
+        try {
+            this.writerConfig = new TableWriter(cairoConfig, configTableName);
+        } catch (CairoException ex) {
+            LOG.error().$("could not open [table=").utf8(configTableName).$("]").$();
+            throw ex;
+        }
+
+        try (TableReader reader = new TableReader(cairoConfig, configTableName)) {
+            final NanosecondClock nanosecondClock = cairoConfig.getNanosecondClock();
+
+            if (reader.size() > 0) {
+                reader.getCursor().getRecord().getLong256(0, idSink);
+                final boolean _enabled = reader.getCursor().getRecord().getBool(1);
+
+                if (enabled != _enabled) {
+                    final TableWriter.Row row = writerConfig.newRow();
+                    row.putLong256(0, idSink);
+                    row.putBool(1, enabled);
+                    row.append();
+                    writerConfig.commit();
+                }
+            } else {
+                final TableWriter.Row row = writerConfig.newRow();
+                row.putLong256(0, nanosecondClock.getTicks(), clock.getTicks(), 0, 0);
+                row.putBool(1, enabled);
+                row.append();
+                writerConfig.commit();
+                reader.reload();
+                reader.getCursor().getRecord().getLong256(0, idSink);
+            }
+        }
+
+        newRow(TelemetryEvent.UP);
+    }
+
+    public int getTableStatus(Path path, CharSequence tableName) {
+        return TableUtils.exists(
+                configuration.getCairoConfiguration().getFilesFacade(),
+                path,
+                configuration.getCairoConfiguration().getRoot(),
+                tableName,
+                0,
+                tableName.length()
+        );
+    }
+
+    private void newRow(short event) {
+        if (enabled) {
+            final TableWriter.Row row = writer.newRow();
+            row.putDate(0, clock.getTicks());
+            row.putShort(1, event);
+            row.append();
+        }
+    }
+
+    private void newRowConsumer(TelemetryTask telemetryRow) {
+        final TableWriter.Row row = writer.newRow();
+        row.putDate(0, telemetryRow.ts);
```

Comment:
ts appears to be in microseconds, putDate() expects millis

Reply:
Done